### ISSUE TITLE
Keyboard events not working on 2D displays 

### ISSUE THREAD

**Issue by curiosity29** [gama-platform/gama.old#3587]
**Describe the bug**
The keyboard event does not work when the type of the display is not specified to be 3d. Also, even with 3d, it only starts working when I click on the display for the first time.

**To Reproduce**
```
model MovingAgents
global
{
	action print { write "r"; }
}
experiment "test" type: gui
{
	output
	{
		display "test" //type: 3d
		{
			event 'r' action: print;
		}
	}
}
```

**Expected behavior**
The action gets triggered whenever the key is pressed.

**GAMA version: 1.9**

**Comment by AlexisDrogoul** [comment]
Please detail your hardware/OS as there has been several issues on keyboard inputs not working with some combinations...

**Comment by curiosity29** [comment]
Processor	Intel(R) Core(TM) i5-9300H CPU @ 2.40GHz   2.40 GHz
Installed RAM	16.0 GB (15.9 GB usable)
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

Edition	Windows 11 Pro
Version	21H2
Installed on	‎8/‎19/‎2022
OS build	22000.1455
Experience	Windows Feature Experience Pack 1000.22000.1455.0

**Comment by AlexisDrogoul** [comment]
Can anyone on Windows 11 / 10 confirm this ? On macOS 13.1, I notice that the Java2D display requires to be explicitly activated (clicking on the tab) before any keyboard event is accepted.

**Comment by lesquoyb** [comment]
I can confirm: java2d on my windows 11 won't run the keyboard events, even after clicking on the display

**Comment by hqnghi88** [comment]
I confirm java2d (2d) on my win 11 wont catch key events, after clicking in the viewport (blank part), but it works if we click on the tab title of the view, or do some layout (maximize, restore view or move view).

**Comment by AlexisDrogoul** [comment]
Hi, the same on macOS : it is only after clicking the tab that the Java2D display receives keyboard events. Does it happen on Linux as well ? 
@RoiArthurB

**Comment by RoiArthurB** [comment]
@AlexisDrogoul Same on Linux with the simple need of activating tab for Java2D and stranger activation condition on OpenGL (random from what I understand it so far... Often need to activate the tab before playing the simulation to have the event active while running the simulation).

---

Tested with the latest alpha pre-released version of GAMA 1.9.0 w/o JDK
